Cubical/Structures/Rng.agda

```diff
@@ -0,0 +1,52 @@
+{-# OPTIONS --cubical --safe #-}
+
+module Cubical.Structures.Rng where
```

Comment:
I don't think we should start with adding Rng but Ring. Is there any reason why you want to do rings without unit?

Reply:
I saw them in Escardo's development, I look up for them and I noticed they exist. Since I want to add as many algebraic structures as possible and rings can be defined from rngs I decided to implement them first.

Reply:
What is the reason to add as many algebraic structures as possible? It seems more reasonable to just add the ones that we actually have instances of in order not to pollute the library with lots of structures that have no examples. I would personally skip also rings and go directly for commutative rings

Reply:
Okay, I will eliminate Rng and change them for Ring, although it may be interesting to have non-commutative rings since someone might want them for matrices? (I don't know if they are in the cubical library, neither if they are interesting in this library) Could you send me an email with the rest of Algebraic structures that may be interesting for the library?

Reply:
One reason for having intermediate structures is that it makes it easier to deal with the day you do need them. If you start with commutative rings and then one day need rings, then presumably commutative rings will need to be redefined (to make use of rings) and a lot of code about commutative rings will need to be updated.
I don’t know if that applies in this situation though, this is just a random thought.

Reply:
Fair enough, let's have rings and commutative rings then. :-) With those structures I feel like we have a good amount of algebraic structures for now and the goal should be to develop some theory about what we have and prove that things are instances of them. 

For the design of the library there are a bunch of things I would like to explore before adding more structures: 

- What to do about the definitions using Records? Nested Sigma types are annoying to use and quite inefficient, but the SIP works best for them. I have an issue about this here: https://github.com/agda/cubical/issues/288

- Can we automate boring algebraic reasoning for the structures that we have? I don't think we can get very far unless we have some automation for helping us with algebraic reasoning.

- Can one use instance arguments to make the operations less annoying to use (see https://github.com/agda/cubical/pull/284/files/3c4c177a4000f748cc3ff6f6c9b77628c2517faa#diff-8b6c459df58628fab692db0ebb992ab8R65). My understanding is that instance arguments are Agda's version of type classes so I guess they can be used for this kind of things? I've never used them though so I might be way off, but without some better inference I don't think the library would be fun to use.

Reply:
I have removed Rng, an added Ring and CommRing. Until we can do the definitions with Records should I define extractors as I have done in Group? I see them really useful. If that
is the case, could I add \<_\> as a syntax that may be imported (so that lives inside a module) from Cubical.Foundation.Structure of the function typ? That will solve that \<_\> must be hidden every time you have two algebraic structures in the same file.

Reply:
Yeah, having extractors is good for now